std/assembly/string.ts

```diff
@@ -393,7 +394,143 @@ export class String {
     return result;
   }
 
+  /* TODO
+   * 1. regex for search
+   * 2. callback function for replacement
+   */
+  replace(search: String, replacement: String): String {
+    assert(this !== null && search !== null && replacement !== null);
+    var len = this.length;
+    var slen = search.length;
+    if (ASC_SHRINK_LEVEL < 1) {
+      if (len <= slen) {
+        if (len < slen) return this;
+        return select<String>(replacement, this, search == this);
+      }
+    }
+    var index = this.indexOf(search);
+    if (~index) {
+      if (ASC_SHRINK_LEVEL >= 1) {
```

Comment:
This forces compilation of `substring` and `concat` even if those are not used anywhere else. Might not be worth it.

Reply:
Done!